zcash_client_sqlite/src/wallet.rs

```diff
@@ -510,14 +535,25 @@ pub(crate) fn add_account<P: consensus::Parameters>(
     }
 
     // Always derive the default Unified Address for the account.
-    let (address, d_idx) = account.default_address(DEFAULT_UA_REQUEST)?;
+    let ua_request = account
+        .uivk()
+        .to_address_request()
+        .and_then(|ua_request| ua_request.intersect(&DEFAULT_UA_REQUEST))
+        .ok_or_else(|| {
+            SqliteClientError::AddressGeneration(AddressGenerationError::ShieldedReceiverRequired)
+        })?;
+    let (address, d_idx) = account.default_address(ua_request)?;
     insert_address(conn, params, account_id, d_idx, &address)?;
 
     // Initialize the `ephemeral_addresses` table.
     #[cfg(feature = "transparent-inputs")]
     transparent::ephemeral::init_account(conn, params, account_id)?;
 
-    Ok(account_id)
+    // Initialize the `ephemeral_addresses` table.
+    #[cfg(feature = "transparent-inputs")]
+    transparent::ephemeral::init_account(conn, params, account_id)?;
+
```

Comment:
Bug: this has been duplicated, which will cause the ephemeral addresses table to be initialized twice for the account. I presume this was introduced by an incorrect merge conflict resolution.
```suggestion
```